### PR TITLE
Added "dirhtml" target

### LINKED ISSUE #6038: yaml config uses "htmldir" instead of "dirhtml"
The Sphinx make target is called "dirhtml", but RTD's [yaml spec ](https://github.com/readthedocs/readthedocs.org/blob/93bcf8d19c367e1088893a4902b15a05d1ad190f/readthedocs/rtd_tests/fixtures/spec/v2/schema.yml#L99) uses "htmldir". RTD should match Sphinx for consistency. It can accept "htmldir" for compatibility, but should only document the correct name.

### PR DESCRIPTION
**Pull request by Blackcipher101**
fixes #6038 
This fixes the issue of Sphinx make target called dirhtml but also make it conistent with RTD

### CODE REVIEW

File: docs/config-file/v2.rst [hunk 1]
```diff
@@ -284,6 +284,7 @@ The builder type for the Sphinx documentation.
 
 :Type: ``string``
 :Options: ``html``, ``htmldir``, ``singlehtml``
+:Options: ``html``, ``dirhtml``, ``singlehtml``
```
Comment (stsewd) [inline docs/config-file/v2.rst]: You should modify the line above, not add a new one. Changing `htmldir` for `dirhtml` (here is fine to replace the old value)

File: readthedocs/rtd_tests/fixtures/spec/v2/schema.yml [hunk 2]
```diff
@@ -97,6 +97,7 @@ sphinx:
   # The builder type for the sphinx documentation
   # Default: 'html'
   builder: enum('html', 'htmldir', 'singlehtml', required=False)
+  builder: enum('html', 'dirhtml', 'singlehtml', required=False)
```
Comment (stsewd) [inline readthedocs/rtd_tests/fixtures/spec/v2/schema.yml]: here you should modify the line above adding the new type in the enum (here we need to keep both, old and new value).

File: docs/config-file/v2.rst (hunk 1 above)
Comment (stsewd) [reply docs/config-file/v2.rst]: Maybe adding a note about the old value was deprecated, so users don't get confused.

Something like: The `htmldir` builder option was renamed to `dirhtml` to use the same name as sphinx. Configurations using the old name will continue working.

**Comment by ericholscher** [comment]
Isn't this backwards incompatible? We'd need to support both and issue a warning.

**Comment by Blackcipher101** [comment]
@stsewd I will work on it

File: readthedocs/rtd_tests/fixtures/spec/v2/schema.yml [hunk 3]
```diff
@@ -96,7 +96,8 @@ python-install-pipfile:
 sphinx:
   # The builder type for the sphinx documentation
   # Default: 'html'
-  builder: enum('html', 'htmldir', 'singlehtml', required=False)
+  builder: enum('html', 'htmldir','dirhtml','singlehtml', required=False)
+
```
Comment (stsewd) [inline readthedocs/rtd_tests/fixtures/spec/v2/schema.yml]:
```suggestion
```

File: docs/config-file/v2.rst [hunk 4]
```diff
@@ -283,7 +283,9 @@ sphinx.builder
 The builder type for the Sphinx documentation.
 
 :Type: ``string``
-:Options: ``html``, ``htmldir``, ``singlehtml``
+:Options: ``html``, ``dirhtml``, ``singlehtml``   
+..note::
+   The htmldir builder option was renamed to dirhtml to use the same name as sphinx. Configurations using the old name will continue working.
```
Comment (stsewd) [inline docs/config-file/v2.rst]: This note should be below the line `Default html`

File: readthedocs/rtd_tests/fixtures/spec/v2/schema.yml [hunk 5]
```diff
@@ -96,7 +96,7 @@ python-install-pipfile:
 sphinx:
   # The builder type for the sphinx documentation
   # Default: 'html'
-  builder: enum('html', 'htmldir', 'singlehtml', required=False)
+  builder: enum('html', 'htmldir','dirhtml','singlehtml', required=False)
```
Comment (stsewd) [inline readthedocs/rtd_tests/fixtures/spec/v2/schema.yml]:
```suggestion
  builder: enum('html', 'htmldir', 'dirhtml', 'singlehtml', required=False)
```

File: docs/config-file/v2.rst (hunk 4 above)
Comment (Blackcipher101) [reply docs/config-file/v2.rst]: Done

**Comment by Blackcipher101** [comment]
@stsewd Thanks helping for out

File: docs/config-file/v2.rst [hunk 6]
```diff
@@ -283,8 +283,10 @@ sphinx.builder
 The builder type for the Sphinx documentation.
 
 :Type: ``string``
-:Options: ``html``, ``htmldir``, ``singlehtml``
+:Options: ``html``, ``dirhtml``, ``singlehtml``
```
Comment (stsewd) [inline docs/config-file/v2.rst]:
```suggestion
:Options: ``html``, ``dirhtml``, ``singlehtml``
```

File: docs/config-file/v2.rst [hunk 7]
```diff
@@ -283,8 +283,10 @@ sphinx.builder
 The builder type for the Sphinx documentation.
 
 :Type: ``string``
-:Options: ``html``, ``htmldir``, ``singlehtml``
+:Options: ``html``, ``dirhtml``, ``singlehtml``   
 :Default: ``html``
+..note::
```
Comment (stsewd) [inline docs/config-file/v2.rst]:
```suggestion

.. note::
```

File: docs/config-file/v2.rst [hunk 8]
```diff
@@ -283,8 +283,10 @@ sphinx.builder
...
+   The htmldir builder option was renamed to dirhtml to use the same name as sphinx. Configurations using the old name will continue working.
```
Comment (stsewd) [inline docs/config-file/v2.rst]:
```suggestion
   The ``htmldir`` builder option was renamed to ``dirhtml`` to use the same name as sphinx.
   Configurations using the old name will continue working.
```

**Pull request merged**